Fastfile

```diff
@@ -110,8 +110,7 @@ end
 
 def _build_ipa
   update_app_identifier(xcodeproj: ENV['FL_UPDATE_PLIST_PROJECT_PATH'],
-                        plist_path: ENV['FL_UPDATE_PLIST_PATH'],
-                        app_identifier: ENV['FL_UPDATE_PLIST_APP_IDENTIFIER'])
+                        plist_path: ENV['FL_UPDATE_PLIST_PATH'])
```

Comment:
just to explain this change a little more...

We don't want to pass in the `app_identifier` argument, so it will default to the `FL_UPDATE_APP_IDENTIFIER` environment variable.

The `FL_UPDATE_PLIST_APP_IDENTIFIER` environment variable that was previously being passed in for this argument is actually what's used by the `update_info_plist` call below. So passing it in here seems redundant. *Therefore I expect this change to be backwardly compatible.*

By passing in the environment variable to the method above, both methods result in using the same environment variable. And BOTH methods update the app identifier.

But they work in different ways. The method above (`update_app_identifier`) will update the Project Build Setting(*) if the Info.plist references it. The method below (`update_info_plist`) will overwrite the Info.plist setting, so that it _no longer references the Project Build Setting_.

An issue can arise if you run the build again without reseting the changes. Because the first method only updates the Project Build Setting if the Info.plist references it. Once the Info.plist setting has been overwritten, then the first method will no longer update the Project Build Setting.

Given that the `create_xcconfig` action relies on the Project Build Setting, we must ensure it's updated correctly.

(*) Project Build Setting is referenced from the Info.plist as `["CFBundleIdentifier": "$(PRODUCT_BUNDLE_IDENTIFIER)"]`

Other solutions considered:

* *Reset the GIT status of the files before running subsequent builds with the same repo clone* - we'll still do this, but it was interesting to understand exactly why we had this problem after changing to multiple bundle IDs, whereas we were previously able to safely build one environment after another
* *Update the `create_xcconfig` action to refer to the `FL_UPDATE_PLIST_APP_IDENTIFIER` environment variable that is injected* - this would be further misuse of this specific environment variable.
* *Introduce a custom environment variable to be used by the `create_xcconfig` action* - i'm open to suggestions on this. it would decouple the two actions (`create_xcconfig` and `udpate_app_identifier`) and mean they could be called in different orders. But it would be an extra environment variable that everyone would need to add.

Reply:
OK, turns out that it isn't backwards compatible as currently stands. Consumers do have to change from `FL_UPDATE_PLIST_APP_IDENTIFIER` to use `FL_UPDATE_APP_IDENTIFIER`. So, question is do we
a. make this a breaking change that goes into v5
b. add backwards compatibility with `FL_UPDATE_APP_IDENTIFIER || FL_UPDATE_PLIST_APP_IDENTIFIER` for v4.0.1
c. do both of the above?

Reply:
i think we should make this a breaking change in v5.